inst/templates/ad_adcoeq.R

```diff
@@ -0,0 +1,257 @@
+# simplified usage throughout the program
+adsl_vars <- exprs(TRTSDT, TRTEDT, TRT01P, TRT01A)
+
+# Read in data ----
+# See the "Read in Data" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#readdata)
+
+# Read data
+qs_metabolic <- admiralmetabolic::qs_metabolic
+adsl <- admiral::admiral_adsl
+
+# Convert SAS missing character values to NA
+adcoeq <- qs_metabolic %>%
+  convert_blanks_to_na()
+adsl <- adsl %>%
+  convert_blanks_to_na()
+
+# Merge ADSL variables (stored in `adsl_vars`) needed for ADQS
+adcoeq <- adcoeq %>%
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  )
+
+# Add analysis parameter variables
+adcoeq <- adcoeq %>%
+  mutate(
+    PARAMCD = QSTESTCD,
+    PARAM = QSTEST,
+    PARCAT1 = QSCAT
+  )
+
+# Derive Date/Time and Analysis Day ----
+# See the "Derive/Impute Numeric Date/Time and Analysis Day" vignette section
+# for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#datetime)
+
+# Add questionnaire analysis date (ADT) and treatment start date (TRTSDT)
+adcoeq <- adcoeq %>%
+  derive_vars_dt(new_vars_prefix = "A", dtc = QSDTC) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT))
+
+# Derive visit info ----
+# See the "Visit and Period Variables" vignette for more information:
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html)
+
+# Derive analysis visit (AVISIT, AVISITN)
+adcoeq <- adcoeq %>%
+  mutate(
+    AVISIT = case_when(
+      is.na(VISIT) ~ NA_character_,
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      str_detect(VISIT, "SCREEN") ~ "SCREENING",
+      ADT <= TRTSDT ~ "Baseline",
+      TRUE ~ str_to_title(VISIT)
+    ),
+    AVISITN = case_when(
+      AVISIT == "Baseline" ~ 0,
+      str_detect(AVISIT, "SCREEN") ~ -1,
+      str_detect(VISIT, "WEEK") ~ as.integer(str_extract(VISIT, "\\d+")),
+      TRUE ~ NA_integer_
+    )
+  )
+
+# Derive results ----
+# See the "Derive Results (AVAL, AVALC)" vignette section for more information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#aval)
+
+# Derive analysis result (AVAL, AVALC)
+adcoeq <- adcoeq %>%
+  mutate(
+    # Invert the original scores as they indicate anxiety and CoEQ item 6
+    # is used in calculating the subscale for "Positive Mood"
+    AVAL = if_else(PARAMCD == "COEQ06", 100 - QSSTRESN, QSSTRESN),
+    AVALC = if_else(PARAMCD == "COEQ20", QSORRES, NA_character_)
+  )
+
+
+# Derive summary records ----
+
+# See the "Example 2(Deriving a Summary Record)" vignette section for more
+# information:
+# (https://pharmaverse.github.io/admiral/articles/bds_finding.html#example-2-deriving-a-summary-record)
+
+# For the Control of Eating Questionnaire, four subscales are derived.
+# These subscales are derived as the sum or the average across a subset of the
```

Comment:
Why do we mention "sum" here?

Reply:
I agree, should be removed. @kathrinflunkert  Are these the only 4 parameters that we derive in COEQ ?

Reply:
Yes, the COEQ manual only mentions these 4 derived parameters.